pwiz/data/msdata/MSData.hpp

```diff
@@ -560,6 +567,9 @@ struct PWIZ_API_DECL Spectrum : public SpectrumIdentity, public ParamContainer
     /// set binary data arrays 
     void setMZIntensityPairs(const MZIntensityPair* input, size_t size, CVID intensityUnits);
 
+    /// set sampled noise data arrays
+    void setNoiseInfo(const NoiseDataInfo* input, size_t size);
+    
```

Comment:
With set/swapBinaryDataArray you can get rid of setNoiseInfo. It'll be easy enough to do all the calls from SpectrumList_Thermo.